cmd/list/cluster/cmd.go

```diff
@@ -39,15 +40,34 @@ var Cmd = &cobra.Command{
 	Run:  run,
 }
 
+const clusterCount = 1000
+
 var listAll bool
 
+var accountRoleArn string
+
 func init() {
 	flags := Cmd.Flags()
 	flags.SortFlags = false
 
 	output.AddFlag(Cmd)
 	flags.BoolVarP(&listAll, "all", "a", false, "List all clusters across different AWS "+
 		"accounts under the same Red Hat organization")
+	flags.StringVar(&accountRoleArn, "account-role-arn", "", "List all clusters "+
+		"using the account role identified by the ARN")
+}
+
+func listClustersUsingAccountRole(creator *aws.Creator, runtime *rosa.Runtime) ([]*v1.Cluster, error) {
+	role, err := runtime.AWSClient.GetAccountRoleByArn(accountRoleArn)
+	if err != nil {
+		return []*v1.Cluster{}, err
+	}
+
+	if role == nil {
+		return nil, fmt.Errorf("no Account role with ARN '%s' exists", accountRoleArn)
```

Comment:
In case the role doesn't exist on AWS, do we get this error from this API call `iamClient.GetRole`?
Regardless of that, I wonder if we should do this check and the error handling inside `GetAccountRoleByArn`?

Reply:
Yeah you're right. The error bubbles up already from the `iamClient.GetRole` call if the role doesn't exist, so this check is redundant